api/jreleaser-model-api/src/main/java/org/jreleaser/model/api/extensions/Extension.java

```diff
@@ -34,6 +34,8 @@ public interface Extension extends Domain, EnabledAware {
 
     String getDirectory();
 
+    String getJBang();
```

Comment:
Gradle DSL requires a similar update https://github.com/jreleaser/jreleaser/blob/522d6f9fd73600f07cd1fa54798436ecdc60b287/plugins/jreleaser-gradle-plugin/src/main/groovy/org/jreleaser/gradle/plugin/dsl/extensions/Extension.groovy#L32

Reply:
We would have to decide how to do it as changing the type after the next release would constitute a breaking change. 

What other elements may be required besides `script` and `args`?

Reply:
args itself does not make sense here - as jreleaser does its own init and skips any main method calling.

Reply:
Version should also be specified otherwise the default bundled version will always be chosen.

I can refactor the code later after merge.

Reply:
but thats the same for maven ?